src/compiler/checker.ts

```diff
@@ -138,6 +138,11 @@ namespace ts {
         const silentNeverType = createIntrinsicType(TypeFlags.Never, "never");
 
         const emptyObjectType = createAnonymousType(undefined, emptySymbols, emptyArray, emptyArray, undefined, undefined);
+
+        const emptyTypeLiteralSymbol = createSymbol(SymbolFlags.TypeLiteral | SymbolFlags.Transient | SymbolFlags.Synthesized, "empty");

```

Comment:
the idea was to use this flag for marking anything what was not present in the original source code.
will change